### ISSUE TITLE
Insert mode cursor in vim remains same in nested tmux session

### ISSUE THREAD

**Issue by ayushkumarshah** [samoshkin/tmux-config#40]
Everything works fine except the cursor remains the same when switching from normal mode to the insert mode inside vim in a nested tmux session (i.e. doesn't become vertical rather than flat as in normal mode). However, it works fine if the tmux session is not nested. Is there any fix for this?

**Comment by Xblow** [comment]
With this configuration, if I ssh into a remote server and open tmux with neovim there, my cursor behaves correctly.

**Comment by ayushkumarshah** [comment]
> With this configuration, if I ssh into a remote server and open tmux with neovim there, my cursor behaves correctly.

@Xblow Yeah, it works that way. The problem is in nested tmux sessions (open tmux in local machine, ssh into remote server inside the tmux session and open tmux session in the remote server)

**Comment by Xblow** [comment]
@ayushkumarshah By "using this config" I meant using tmux with this config, within which I'd ssh into a server, shoot up tmux there and launch neovim.
If I use F12 to switch off root tmux key bindings then cursor also behaves correctly.
I don't know if this would be the case with original vim (with custom cursor) as I did not check that (and I am not a maintainer). Which flavour of vim do you use?
Hope you'll get this issue resolved :)

**Comment by ayushkumarshah** [comment]
> By "using this config" I meant using tmux with this config, within which I'd ssh into a server, shoot up tmux there and launch neovim.
> If I use F12 to switch off root tmux key bindings then cursor also behaves correctly.

Ohh, I see. I don't know what is causing the issue in my case.

> Which flavour of vim do you use?

I use neovim.
